lib/bolt/inventory/inventory2.rb

```diff
@@ -138,6 +157,7 @@ def update_target(target)
         set_vars_from_hash(target.name, data['vars']) unless @target_vars[target.name]
         set_facts(target.name, data['facts']) unless @target_facts[target.name]
         data['features']&.each { |feature| set_feature(target, feature) } unless @target_features[target.name]
+        data['config'] = config_plugin(data['config'])
```

Comment:
I think there may be a bug here/`@groups.data_for()` where we deep merged config values from two sources one value is a hash of raw config data and the other is a plugin_lookup hash. Where the `_plugin` will get added to the raw data causing it to be evaluated as a plugin and throw away the raw data regardless of precedence order.

Reply:
I think with the updated implementation that converts every instance of a `{'_plugin' => ... }` hash to a concurrent delay instance before instantiating a `Group2` object we avoid this issue with `deep_merge`. https://github.com/puppetlabs/bolt/pull/994/commits/21fd90e2a2748cf6c638eba380c82faa47bc35d3#diff-8f5017c5ca9a2d12a104805c2a404110R23